docs/tools.md

```diff
@@ -0,0 +1,190 @@
+
+### Proving stuff
+
+You can now check your first proof.
+In a file called `test.lean`, type:
+```lean
+lemma zero_max (m : ℕ) : max 0 m = m :=
+max_eq_right (nat.zero_le m)
+```
+Then you can ask Lean to check this proof by running `lean test.lean`. 
+Lean will think for a second (most of it spent in intialization) and return control to you, without outputing anything.
+That's Lean's idea of a dignified triumph.
+If you start messing up with the file, say deleting the final `m` on the last line,
+Lean will output an error message.
+That's 99.9% of what Lean and its supporting tool do,
+but the remaining 0.1% will make your life much easier. 
+
+Before turning to that, you need to understand why Lean didn't ask for the definition of `ℕ`, `max`, `max_eq_right` and `nat.zero_le`.
+That's because all those are defined in the part of Lean's core library that is automatically loaded by default.
+This default library lives in the `lib` folder you saw after downloading Lean, 
+precisely in `lib/lean/library/init/`.
+You can prevent loading it by starting your `test.lean` with the line `prelude`.
+Then everything will fall apart. 
+Lean will not know about `ℕ`, `max`, or even the equality sign!
+And it won't attempt to read the proof of a statement it couldn't understand.
+So you should not keep that `prelude` line.
+
+
+###  Working with several files
+
+Your Lean work has grown quite a bit since you installed Lean,
+so let's start a second file, `test2.lean` that should build on the knowledge gathered in `test.lean`.
+In `test2.lean`, put an important special case of our `zero_max` lemma:
+```lean
+lemma zero_max_one : max 0 1 = 1 :=
+zero_max 1
+```
+Trying to compile it with `lean test2.lean` fails: Lean complains it doesn't know about
+our `zero_max` lemma.
+You need to tell Lean that `test2.lean` relies on `test.lean`. 
+So add at the top of `test2.lean` the line `import test`.
+Now Lean complains it cannot find the file `test` in `LEAN_PATH`.
+You can ask Lean where it searches for files by running `lean --path`, and paying attention
+to the lines in the "path" list.
+Notice this list does not contain the folder where `test.lean` is sitting.
+That all caps name in the error message suggests setting an environment variable called `LEAN_PATH` could help.
+Indeed you can run:
+```bash
+LEAN_PATH=path_to_our_lean_install_folder/lib/lean/library/:path_to_folder_containing_test lean test2.lean 
+```
+and this succeeds.
+Note that omitting the first part would have brought you to the prelude situation where Lean does not know
+about natural numbers or equality.
+
+Setting this `LEAN_PATH` variable is clearly annoying.
+There is a much better way, and actually you should *never* set this environment variable.
+Now create, next to `test.lean` and `test2.lean`, a file `leanpkg.path` containing:
+```
+builtin_path
+path .
+```
+That's it, each time you'll invoke Lean from within this folder or one of its subfolders,
```

Comment:
I think it would be good to mention at this point that the way `builtin_path` is resolved is by going to `../library` and `../lib/lean/library` relative to the location of the executing lean process. Usually lean will be at `lean-install-dir/bin/lean` and so this will go to either `lean-install-dir/bin/library` (which is used when you build the repo) or  `lean-install-dir/bin/lib/lean/library` (which is used when you use the bare binary distribution).